lib/readthis/cache.rb

```diff
@@ -352,6 +355,10 @@ def extract_options!(array)
     end
 
     def merged_options(options)
```

Comment:
You are correct, that is what I'm suggesting. There isn't any plausible way to guard against calling the `to_i` on every invocation of `merged_options`, so there isn't much point to defining the same logic in two places.

Regarding the performance aspects of it, this is precisely why I opted *not* to support casting duration like values to an integer. I'm still leaning that way, but a benchmark could be in order to see if there is any appreciable difference. You can run a modified version of `benchmarks/generic.rb` before and after the change to check things out.